v2/internal/identity/credential_provider.go

```diff
@@ -334,6 +335,42 @@ func (c *credentialProvider) newCredentialFromSecret(secret *v1.Secret) (*Creden
 		}, nil
 	}
 
+	// This authentication type is similar to user assigned managed identity authentication combined with client certificate
+	// authentication. As a 1st party Microsoft application, one has access to pull a user assigned managed identity's backing
+	// certificate information from the MSI data plane. Using this data, a user can authenticate to Azure Cloud.
+	if userAssignedCredentialsPath, hasUserAssignedCredentials := secret.Data[config.AzureUserAssignedIdentityCredentials]; hasUserAssignedCredentials {
+		// Default to AzurePublic
+		options := azcore.ClientOptions{
+			Cloud: c.cloud,
+		}
+
+		// Override the default cloud configuration if the cloud configuration was configured in the secret
+		if string(secret.Data[config.ResourceManagerAudience]) != "" {
```

Comment:
I don't think you need this section on 348, because it's also about selecting cloud. Just pass `options` above to the `NewUserAssignedIdentityCredentials`